OpenRA.Game/MiniYaml.cs

```diff
@@ -435,18 +435,41 @@ static List<MiniYamlNode> MergePartial(List<MiniYamlNode> existingNodes, List<Mi
 
 			var existingDict = existingNodes.ToDictionaryWithConflictLog(x => x.Key, "MiniYaml.Merge", null, x => $"{x.Key} (at {x.Location})");
 			var overrideDict = overrideNodes.ToDictionaryWithConflictLog(x => x.Key, "MiniYaml.Merge", null, x => $"{x.Key} (at {x.Location})");
-			var allKeys = existingDict.Keys.Union(overrideDict.Keys);
 
-			foreach (var key in allKeys)
+			foreach (var node in existingNodes.Concat(overrideNodes))
 			{
-				existingDict.TryGetValue(key, out var existingNode);
-				overrideDict.TryGetValue(key, out var overrideNode);
-
-				var loc = overrideNode?.Location ?? default;
-				var comment = (overrideNode ?? existingNode).Comment;
-				var merged = (existingNode == null || overrideNode == null) ? overrideNode ?? existingNode :
-					new MiniYamlNode(key, MergePartial(existingNode.Value, overrideNode.Value), comment, loc);
-				ret.Add(merged);
+				// Append Removal nodes to the result.
+				// Therefore: we know the remainder of the loop deals with plain nodes.
+				if (node.Key.StartsWith("-", StringComparison.Ordinal))
+				{
+					ret.Add(node);
+					continue;
+				}
+
+				// If no previous node with this key is present, it is new and can just be appended.
+				var previousNodeIndex = ret.FindLastIndex(n => n.Key == node.Key);
+				if (previousNodeIndex == -1)
+				{
+					ret.Add(node);
+					continue;
+				}
+
+				// A Removal node is closer than the previous node.
```

Comment:
```suggestion
				// A removal node is above the previous node.
```

Reply:
It is actually "below"/"under"/"after". Between the previous such node and the current one.